api/opds.py

```diff
@@ -158,6 +159,20 @@ def active_licensepool_for(self, work):
             return super(
                 CirculationManagerAnnotator, self).active_licensepool_for(work)
 
+    def visible_delivery_mechanisms(self, licensepool):
+        """Filter the given `licensepool`'s LicensePoolDeliveryMechanisms
+        to those with content types that are not hidden.
+        """
+        hidden = self.hidden_content_types
+        for lpdm in licensepool.delivery_mechanisms:
+            mechanism = lpdm.delivery_mechanism
+            if not mechanism:
+                # This shouldn't happen, but just in case.
+                continue
+            if mechanism.content_type in hidden:
+                continue
+            yield lpdm
```

Comment:
Just curious, why the choice to use a generator for this?

Reply:
Just personal preference I guess. Most of the time I find it easier to write a generator than to keep track of a list inside of a method.